src/core/structure/sheet.ts

```diff
@@ -272,6 +275,64 @@ export default class Sheet {
     return true;
   }
 
+  private updateReferenceSymbolsForGroup(
+    group: CellGroup<ColumnKey | RowKey>,
+    from: number,
+    to: number,
+  ) {
+    for (const [oppositeKey, cellKey] of group!.cellIndex) {
+      const cell = this.cellData.get(cellKey)!;
+      if (!cell.referencesIn) continue; // Only cells with incoming references are affected.
+
+      // Convert the information on how the group is shifted into coordinates.
+      // from and to can refer to either a column or row position depending on the group type.
+      const oppositeGroupPos =
+        group instanceof Column
+          ? this.rows.get(oppositeKey as RowKey)!.position
+          : this.columns.get(oppositeKey as ColumnKey)!.position;
+
+      const fromCoord: Coordinate = {
+        column: group instanceof Column ? from : oppositeGroupPos!,
+        row: group instanceof Row ? from : oppositeGroupPos!,
+      };
+
+      const toCoord: Coordinate = {
+        column: group instanceof Column ? to : oppositeGroupPos!,
+        row: group instanceof Row ? to : oppositeGroupPos!,
+      };
+
+      this.updateCellReferenceSymbols(cell, fromCoord, toCoord);
+    }
+  }
+
+  private updateCellReferenceSymbols(
```

Comment:
Does a case exist where updating reference symbols can create a circular reference? The updating is working as intended in the tests you added. In the first test for example, moving the row results in A1 -> "=A2", A2 -> "=B2", B2 = "100" which is as expected.

Reply:
Right, it's not an updating reference symbols issue at all! So there is no issue related to merging this PR

But where should the circular reference be checked if we move a row/column to a place where it creates a circular reference, is this taken care of somewhere else? Because it seems that the cell state is not updated in this kind of case

Reply:
> But where should the circular reference be checked if we move a row/column to a place where it creates a circular reference, is this taken care of somewhere else?

Moving a row or column doesn't modify the internal reference structure, so it shouldn't be possible to create a circular reference.